packages/kotti-ui/source/kotti-field-file-upload/components/DropArea.vue

```diff
@@ -0,0 +1,276 @@
+<template>
+	<div
+		ref="dropAreaRef"
+		:class="classes"
+		:data-test="dataTest ? `${dataTest}.dropArea` : undefined"
+		:tabIndex="_tabIndex"
+		@click.prevent="onClick"
+		@dragleave.prevent.stop="isDragging = false"
+		@dragover.prevent.stop="isDragging = true"
+		@drop.prevent.stop="onDrop"
+		@mouseenter.prevent.stop="isHovering = true"
+		@mouseleave.prevent.stop="isHovering = false"
+	>
+		<div class="kt-field-file-upload__drop-area__description">
+			<i class="yoco" v-text="_icon" />
+			<a v-text="translations.text.clickToUpload" />
+			{{ translations.text.dragAndDrop }}
+		</div>
+		<div
+			v-if="showInformation"
+			class="kt-field-file-upload__drop-area__information"
+		>
+			{{ informationText }}
+			<a
+				v-if="infoUrl"
+				:href="infoUrl"
+				rel="noopener noreferrer"
+				target="_blank"
+				@click.stop
+				v-text="translations.text.learnMore"
+			/>
+		</div>
+		<input
+			v-show="false"
+			ref="uploadInputRef"
+			:accept="accept"
+			:data-test="dataTest ? `${dataTest}.input` : undefined"
+			:multiple="allowMultiple"
+			type="file"
+			@change="onSelect"
+			@click.stop
+		/>
+		<div
+			v-if="$slots.content"
+			class="kt-field-file-upload__drop-area__content-slot"
+			@click.stop="onContentSlotClick"
+		>
+			<slot name="content" />
+		</div>
+		<div
+			v-if="$slots.footer"
+			class="kt-field-file-upload__drop-area__footer-slot"
+			@mouseenter.prevent.stop="isHovering = false"
+			@mouseleave.prevent.stop="isHovering = true"
+		>
+			<slot name="footer" />
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import {
+	computed,
+	defineComponent,
+	onBeforeMount,
+	onUnmounted,
+	ref,
+} from '@vue/composition-api'
+
+import { useTranslationNamespace } from '../../kotti-i18n/hooks'
+import { KEY_ENTER, KEY_SPACE } from '../constants'
+import { KottiFieldFileUpload } from '../types'
+import { formatAccept, formatFileSize, getFileItem } from '../utils'
+import { validateSelection } from '../validators'
+
+export default defineComponent({
+	name: 'DropArea',
+	props: {
+		accept: { default: null, type: String },
+		acceptLabel: { default: null, type: String },
+		allowMultiple: { default: true, type: Boolean },
+		dataTest: { default: null, type: String },
+		icon: { default: null, type: String },
+		infoUrl: { default: null, type: String },
+		isDisabled: { default: false, type: Boolean },
+		maxFileSize: { default: null, type: Number },
+		tabIndex: { default: 0, type: Number },
+	},
+	setup(
+		props: Pick<
+			KottiFieldFileUpload.PropsInternal,
```

Comment:
Would it have made more sense to just scope the props within the namespace of the KottiFieldFileUpload

I will leave the example comment on the types file